src/query.ts

```diff
@@ -8,11 +8,22 @@ import { ChangeEvent } from './index';
  */
 export type QueryOptions = {
     filter: any,
+    options: FilterOptions,
```

Comment:
We can avoid `options.options` if we pass each option field separately. Also better make them optional with the `?` suffix.